src/controllers/cartController.ts

```diff
@@ -1,144 +1,206 @@
+    }
+
+    const { userId, productId } = req.body;
+
+    const cart = await CartsModel.findOne({
+      where: { user_id: userId, product_id: productId },
+    });
+
+    if (!cart) {
+      return res.status(404).json({ error: 'Product not found in the user\'s cart' });
     }
-  };
-  
-  
\ No newline at end of file
+
+    await CartsModel.destroy({
+      where: { user_id: userId, product_id: productId },
+    });
+
+    return res.status(200).json({ message: 'Product removed from cart successfully' });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ error: 'Internal server error' });
+  }
+};
+
+export const clearCart = async (req: Request, res: Response) => {
+  try {
+    const userId = parseInt(req.params.id);
+
+    if (isNaN(userId)) {
+      return res.status(400).json({ error: 'Invalid user ID' });
+    }
+
+    await CartsModel.destroy({
+      where: { user_id: userId },
+    });
+
+    return res.status(200).json({ message: 'Cart cleared successfully' });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ error: 'Internal server error' });
+  }
+};
+
+export const decreaseQuantity = async (req: Request, res: Response) => {
+  try{
+    const value = (req.body);
+    const { productId, userId } = value;
+    const cart = await CartsModel.findOne({
+      where: { user_id: userId, product_id: productId },
+    });
+
+    if (!cart) {
+      return res.status(404).json({ error: 'Product not found in the user\'s cart' });
+    }
+
+    const currentQuantity: number = cart.get('quantity') as number;
+    if (currentQuantity <= 1) {
+      return res.status(400).json({ error: 'Product quantity cannot be decreased below 1' });
+    }
+
+    await cart.update({ quantity: currentQuantity - 1 });
+
+    const updatedCartItem = await CartsModel.findOne({
+      attributes: { exclude: ['id'] }, 
+      where: { user_id: userId, product_id: productId },
+    });
+
+    return res.status(200).json({ message: 'Product quantity decreased successfully', updatedCartItem });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ error: 'Internal server error' });
+  }
+};
+
+export const increaseQuantity = async (req: Request, res: Response) => {
+  try {
+    const value = req.body;
+    const { productId, userId } = value;
+
+    const cart = await CartsModel.findOne({
+      where: { user_id: userId, product_id: productId },
+    });
+
+    if (!cart) {
+      return res.status(404).json({ error: 'Product not found in the user\'s cart' });
+    }
+
+    const currentQuantity: number = cart.get('quantity') as number;
+    await cart.update({ quantity: currentQuantity + 1 });
+
+    const updatedCartItem = await CartsModel.findOne({
+      attributes: { exclude: ['id'] },
+      where: { user_id: userId, product_id: productId },
+    });
+
+    return res.status(200).json({ message: 'Product quantity increased successfully', updatedCartItem });
```

Comment:
return message:"success" message has to be a known string because frontend will check on it if(data.message=="success") {
 // code
}

Reply:
DONE